app/controllers/api/v1/public_api/experience_rewards_controller.rb

```diff
@@ -0,0 +1,24 @@
+class API::V1::PublicAPI::ExperienceRewardsController < API::V1::PublicAPI::APIController
+  def index
+    rewards = ExperienceReward.active.order(:cost)
+
+    render json: {rewards: API::ExperienceRewardBlueprint.render_as_json(rewards, view: :normal, current_user: current_user)}, status: :ok
+  end
+
+  def claim
+    reward = ExperienceReward.find_by(uuid: params[:experience_reward_id])
```

Comment:
We already have a catch here: https://github.com/talentprotocol/web-dapp/blob/b34915e15ed8b354f49b3c2da636d38d1889ca48/app/controllers/api/v1/public_api/api_controller.rb#L60

You can remove the if/else.

```suggestion
    reward = ExperienceReward.find_by!(uuid: params[:experience_reward_id])
```

Reply:
This allows me to send a specific message on this endpoint to help the frontend show a more meaningful message. (the message itself will still be improved)